File: lab_12/activity_logger.py
import datetime

def log_activity():
    with open("log.txt", "a") as log_file:
        now = datetime.datetime.now()
        log_file.write(f"User started the program at {now.strftime('%Y-%m-%d %H:%M:%S')}\n")
        print("[บันทึกการใช้งานลงไฟล์เรียบร้อย]")

def read_log():
    try:
        with open("log.txt", "r") as log_file:
            print("\n[อ่านข้อมูลจาก log.txt]")
            for line in log_file:
                print(line, end="")
    except FileNotFoundError:
        print("Unable to open log file.")

if __name__ == "__main__":
    log_activity()
    read_log()

File: lab_12/activity_logger.cpp
#include <iostream>
#include <fstream>
#include <ctime>
using namespace std;

void logActivity() {
    ofstream logFile("log.txt", ios_base::app);
    if (logFile.is_open()) {
        time_t now = time(nullptr);
        logFile << "User started the program at " << ctime(&now);
        logFile.close();
        cout << "[บันทึกการใช้งานลงไฟล์เรียบร้อย]\n";
    } else {
        cerr << "Unable to open log file.\n";
    }
}

void readLog() {
    ifstream logFile("log.txt");
    if (logFile.is_open()) {
        cout << "\n[อ่านข้อมูลจาก log.txt]\n";
        string line;
        while (getline(logFile, line)) {
            cout << line << "\n";
        }
        logFile.close();
    } else {
        cerr << "Unable to open log file.\n";
    }
}

int main() {
    logActivity();
    readLog();
    return 0;
}
